query-node/mappings/content-directory/entity/remove.ts

```diff
@@ -48,49 +48,28 @@ async function removeLicense(db: DB, where: IWhereCond): Promise<void> {
   const record = await db.get(LicenseEntity, where)
   if (record === undefined) throw Error(`License not found`)
 
-  const { knownLicense, userdefinedLicense } = record
-  let videos: Video[] = []
-
-  if (knownLicense) {
-    videos = await db.getMany(Video, {
-      where: {
-        license: {
-          isTypeOf: 'KnownLicense',
-          code: knownLicense.code,
-          description: knownLicense.description,
-          name: knownLicense.name,
-          url: knownLicense.url,
-        },
-      },
-    })
-  }
-  if (userdefinedLicense) {
-    videos = await db.getMany(Video, {
-      where: { license: { isTypeOf: 'UserDefinedLicense', content: userdefinedLicense.content } },
-    })
+  if (record.videolicense) {
+    // Remove all the videos under this license
+    record.videolicense.map(async (v) => await removeVideo(db, { where: { id: v.id } }))
```

Comment:
Does it make sense to await for all `removeVideo` operations to complete before removing the License Entity ? Could there be a foreign key violation if we do not?

```suggestion
    await Promise.all(record.videolicense.map(async (v) => await removeVideo(db, { where: { id: v.id } })))
```

Reply:
Yes, otherwise it will not be possible to delete the license since videos have references. But this code (removing videos) should never get executed because runtime enforces reference counter to be zero so they are removed in an order that key violation would never happen. I was just keeping them around maybe would be better to remove them.

Reply:
Okay thanks for clarifying. So when you say code should never get executed, that is generally an indication that you should not actually have any code there and as you suggested remove it (because you are trying to fix a mistake in the runtime) and instead have an assertion that `record.videolicense` is an empty array instead.

Reply:
You could ofcourse just remove the code completely, and the processor would fail with an ugly message about key violation. Having the assertion with a good message will make it easier to debug the error if it ever happens in the future.

Reply:
Done.